### PR TITLE
Rename all references of bioformats to files

### PR DESCRIPTION
**Pull request by rleigh-dundee**
Complete the renaming following the split.

Note that there is an unresolved dichotomy in that "OME Files" is used to name both the "ome-files" library in this repository *and* refer to the overarching name used to refer to all the combined components in the binary release.  This makes the documentation ambiguous.  Note that it's also not the top-level component; that's currently qtwidgets and the view program based upon it.  Any other components we add which use ome-files will also make this an intermediate component rather than the "top" one.  As discussed on Friday, I'm particularly unhappy with this for the shared library since it makes for an illogical and inconsistent naming scheme.  I'd rather we used it to refer to one thing only, which is the overarching name for the release.  If we aren't willing to reconsider naming the shared library, then we might want to consider renaming the overarching name instead.

All references to `[ome-]bioformats`, `bio-formats`, `[OME]BioFormats`, `[OME-]Bio-Formats` or `[OME[-_]]BIOFORMATS` are now `ome-files`, `ome-files`, `[OME][Files]`, `[OME-]Files` or `[OME[-_]]FILES`, respectively
All references to `bf-` or `bf_` are now `ome-files` or `ome_files_`, respectively

There are some additional minor cleanups for the split components as well.

See also:
  https://github.com/ome/ome-common-cpp/pull/10
  https://github.com/openmicroscopy/bioformats/pull/2264

--------

Testing:

The builds should remain green.
The `bf-test` command is now `ome-files`; it should work identically to before

### CODE REVIEW

**Comment by joshmoore** [comment]
> Note that it's also not the top-level component; that's currently qtwidgets and the view program based upon it.

I see qtwidgets as being a good candidate for decoupling.

> Any other components we add which use ome-files will also make this an intermediate component rather than the "top" one.

Can you give an example of other higher-level artifacts you see as being in this repo?

> All references to [ome-]bioformats, bio-formats, [OME]BioFormats, [OME-]Bio-Formats or [OME[-]]BIOFORMATS are now ome-files, ome-files, [OME][Files], [OME-]Files or [OME[-]]FILES respectively
> All references to bf- or bf_ are now ome-files or ome_files_ respectively

:+1:

**Comment by rleigh-dundee** [comment]
@sbesson qtwidgets could have been split out--I did suggest it--but it was decided not to.  I can't give any examples for other higher-level artefacts because right now we don't have anything.  Future additions would be tools and libraries building on the basic functionality already provided.

But these are kind of unrelated to the problem at hand.  Whether we split out qtwidgets or add or do not add other components is irrelevant to the fact that the naming scheme this introduces is inconsistent and confusing: we use the same name to refer to two completely different things.

**Comment by joshmoore** [comment]
> qtwidgets could have been split out--I did suggest it--but it was decided not to.

Agreed, and definitely not a criticism at all. For the moment, the state of the decoupling is great. But if qtwidgets does end up leading to confusion, we can certainly reconsider later.

> .. the fact that the naming scheme this introduces is inconsistent and confusing: we use the same name to refer to two completely different things.

Your concern is understood. I'd say let's keep along the lines that you've started the renaming, and review when we have a few concrete artifacts that we can compare with others. Thanks!

**Comment by rleigh-dundee** [comment]
Green in https://ci.openmicroscopy.org/view/DEV/job/BIOFORMATS-CPP-DEV-merge-win-superbuild/139/ and https://ci.openmicroscopy.org/view/DEV/job/BIOFORMATS-CPP-DEV-merge-superbuild/121/ and https://ci.openmicroscopy.org/view/DEV/job/BIOFORMATS-CPP-DEV-merge/96/

**Comment by mtbc** [comment]
So I'm guessing that the BF CPP DEV merge jobs do consume open PRs from this repository. :smiley:

**Comment by mtbc** [comment]
Using the merge build artifact from gretzky,
```
$ /tmp/ome-files-5.2.0-Release-CentOS6.7-x86_64/bin/ome-files info multi-channel.ome.tif   
Image: multi-channel.ome.tif
Using reader: OME-TIFF (Open Microscopy Environment TIFF)
[2016-03-02 09:57:37.110719] [0x00007f26d72c0780] [warning] Ignoring null UUID object when retrieving filename
[2016-03-02 09:57:37.110772] [0x00007f26d72c0780] [warning] Ignoring null UUID object when retrieving value
[2016-03-02 09:57:37.110813] [0x00007f26d72c0780] [warning] Image ID: Image:0 missing plane #1
Reader setup took 00:00:00.268413

Filename = "/home/mtbc/ome/data_repo/test_images_good/ome-tiff/multi-channel.ome.tif"
Used files = ["/home/mtbc/ome/data_repo/test_images_good/ome-tiff/multi-channel.ome.tif"]

Reading core metadata
Series count = 1

Series #0:
	Image count = 3
	RGB = [false, false, false] ([1, 1, 1]) 
	Interleaved = false
	Indexed = false
	Width = 439
	Height = 167
	SizeZ = 1 (effectively 1)
	SizeT = 1 (effectively 1)
	SizeC = 3 (effectively 3)
	Thumbnail size = 128 × 48
	Endianness = little
	DimensionOrder = XYZCT (certain)
	PixelType = uint8
	Bits per Pixel = 8
	MetadataComplete = true
	ThumbnailSeries = false

No global metadata
```
looks good. :+1:

**Comment by rleigh-dundee** [comment]
@mtbc Yes, it's all integrated into the CI jobs.

**Comment by mtbc** [comment]
I'm not sure how badly the documentation will need the two things that are possibly "OME Files" deconfused, how often context will make the referent clear nor what helpful words to use when it doesn't, but I can certainly see that having the *environment variables* for the general top-level thing and the specific component differently named could be much appreciated by some users (depending on anticipated installation, etc.) and most easily deconfused if not first confused.

**Comment by joshmoore** [comment]
Discussion outlined under https://trello.com/c/3DRYnnyk/27-m3-units-versioning . Current proposal is @mtbc's "OME Files Bundle"

**Comment by rleigh-dundee** [comment]
This should be complete now, with the final change here plus the three PRs linked above.  Is there anything else you require here?

File: docs/sphinx/conf.py.in [hunk 1]
```diff
@@ -40,7 +40,7 @@ from datetime import datetime
 extensions = ['sphinx.ext.extlinks', 'edit_on_github']
 
 ## Configuration for the edit_on_github extension
-edit_on_github_project = 'openmicroscopy/bioformats'
+edit_on_github_project = 'ome/files'
```
Comment (joshmoore) [inline docs/sphinx/conf.py.in]: Does this imply github.com/ome/files?

Comment (rleigh-dundee) [inline docs/sphinx/conf.py.in]: Probably.  This was likely from a blanket sed over the whole repo.  I can rename to ome/ome-files-cpp.

As a followup, does the repo need to be `ome-files-cpp` or could it just be `ome-files` and potentially have the java stuff merged in down the line?  I wasn't sure why we bothered with the suffix here; my understanding was that ome-common-cpp was a special-case due to the clash with java.

Comment (joshmoore) [inline docs/sphinx/conf.py.in]: @rleigh-dundee : I had the same thought re: the suffix. I think the last time we discussed we said that worst case we would join the git history some future `ome-files-java` to create a single `ome-files` if need be.

**Comment by sbesson** [comment]
Thanks. https://ci.openmicroscopy.org/view/Failing/job/BIOFORMATS-CPP-DEV-merge-trigger/134/ is green now. Other naming discussions might happen (like the executable name), but we can tackle them in follow-up PRs.

**Pull request merged**